frame/election-provider-support/src/lib.rs

```diff
@@ -543,7 +547,11 @@ impl<
 	fn solve(
 		winners: usize,
 		targets: Vec<Self::AccountId>,
-		voters: Vec<(Self::AccountId, VoteWeight, impl IntoIterator<Item = Self::AccountId>)>,
+		voters: Vec<(
+			Self::AccountId,
+			VoteWeight,
+			impl IntoIterator<Item = Self::AccountId> + Clone,
```

Comment:
why the added `Clone`?

Reply:
I am cloning `voters` is there a way to avoid adding that trait while still being able to clone `voters`?

Reply:
Sorry I got you now, you were commenting on the other 2 algorithm, I removed the trait.